tap_zendesk/streams.py

```diff
@@ -137,7 +137,9 @@ def raise_or_log_zenpy_apiexception(schema, stream, e):
     # it doesn't have access.
     if not isinstance(e, zenpy.lib.exception.APIException):
         raise ValueError("Called with a bad exception type") from e
-    if json.loads(e.args[0])['error']['message'] == "You do not have access to this page. Please contact the account owner of this help desk for further help.":
+
+    error = json.loads(e.args[0]).get('error')
+    if isinstance(error, dict) and error.get('message', None) == "You do not have access to this page. Please contact the account owner of this help desk for further help.":
```

Comment:
Please add a comment here about this change.

Reply:
> Please add a comment here about this change.

Added a comment

Reply:
Is it worth converting the dict to class?

Reply:
@KrisPersonal, here we are just checking that the error is of type dictionary or not. As we are using Zenpy SDK, it throws different errors according to different issues. Moreover, the error retrieved can be of type dictionary or a string in some cases. Hence, in this `if isinstance(error, dict)` we are just verifying that if this error retrieved is of type dictionary, if yes, then only we'll go further and get the message through `error.get('message', None)`. We are not converting a dict to class.